app/assets/javascripts/discourse/app/lib/site-setting-filter.js

```diff
@@ -0,0 +1,146 @@
+import { bind } from "discourse-common/utils/decorators";
+import I18n from "discourse-i18n";
+
+export default class SiteSettingFilter {
+  constructor(siteSettings) {
+    this.siteSettings = siteSettings;
+  }
+
+  filterSettings(filter, opts = {}) {
+    opts.maxResults ??= 100;
+    opts.onlyOverridden ??= false;
+
+    return this.performSearch(filter, opts);
+  }
+
+  @bind
+  performSearch(filter, opts) {
```

Comment:
This can all likely be cleaned up more, but I am just moving it for now

Reply:
Good call. This isn't really reviewable in the current state, so better to lift-and-shift since we Know It Works™. 🙂